### PR TITLE
Image upload

### PR DESCRIPTION
**Pull request by sakibrahmanchy**

### CODE REVIEW

File: app/Repository/Repositories/ImageRepository.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+class ImageRepository extends BaseRepository implements ImageRepositoryInterface {
+    protected $model;
+    protected $disk = 's3';
+
+    public function __construct(Image $model)
+    {
+        $this->model = $model;
+    }
+
```
Comment (sakibrahmanchy) [inline app/Repository/Repositories/ImageRepository.php:18]: extra space here

File: routes/api.php [hunk 2]
```diff
@@ -33,6 +33,8 @@
         return $request->user();
     });
 
+
```
Comment (sakibrahmanchy) [inline routes/api.php]: same here

File: app/Http/Controllers/ImageController.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
+<?php
+
```
Comment (alshahriar042) [inline app/Http/Controllers/ImageController.php:2]: Extra Space Here

File: app/Providers/RepositoryServiceProvider.php [hunk 4]
```diff
@@ -22,6 +24,7 @@ public function register()
         $this->app->bind(EloquentRepositoryInterface::class, BaseRepository::class);
         $this->app->bind(UserRepositoryInterface::class, UserRepository::class);
         $this->app->bind(FollowRepositoryInterface::class, FollowRepository::class);
+        $this->app->bind(ImageRepositoryInterface::class, ImageRepository::class);
```
Comment (fahimfoysal2) [inline app/Providers/RepositoryServiceProvider.php:27]: Why not bind in boot ?

Comment (sakibrahmanchy) [reply app/Providers/RepositoryServiceProvider.php:27]: Always bind stuffs on register.

**Review by fahimfoysal2 — APPROVED** [review]

**Pull request merged**